openedx_authz/tests/test_enforcement.py

```diff
@@ -0,0 +1,446 @@
+    def _load_policy(self, policy: list[str]) -> None:
+        """
+        Load policy rules into the Casbin enforcer.
+
+        This method clears any existing policies and loads the provided policy rules
+        into the appropriate policy stores (p for policies, g for role assignments,
+        g2 for action groupings).
+
+        Args:
+            policy (list[str]): List of policy rules where each rule is a
+                list starting with the rule type ('p', 'g', or 'g2') followed by
+                the rule parameters.
+
+        Raises:
+            ValueError: If a policy rule has an invalid type (not 'p', 'g', or 'g2').
+        """
+        self.enforcer.clear_policy()
+        for rule in policy:
+            if rule[0] == "p":
+                self.enforcer.add_named_policy("p", rule[1:])
+            elif rule[0] == "g":
+                self.enforcer.add_named_grouping_policy("g", rule[1:])
+            elif rule[0] == "g2":
+                self.enforcer.add_named_grouping_policy("g2", rule[1:])
+            else:
+                raise ValueError(f"Invalid policy rule: {rule}")
+
+    def _test_enforcement(self, policy: list[str], request: AuthRequest) -> None:
+        """
+        Helper method to test enforcement and provide detailed feedback.
+
+        Args:
+            policy (list[str]): A list of policy rules to load into the enforcer
+            request (AuthRequest): An authorization request containing all necessary parameters
+        """
+        self._load_policy(policy)
+        subject, action, scope = request["subject"], request["action"], request["scope"]
+        result = self.enforcer.enforce(subject, action, scope)
+        error_msg = f"Request: {subject} {action} {scope}"
+        self.assertEqual(result, request["expected_result"], error_msg)
+
+
+@ddt
+class SystemWideRoleTests(CasbinEnforcementTestCase):
+    """
+    Tests for system-wide roles with global access permissions.
+
+    This test class verifies that users assigned to system-wide roles (with global scope "*")
+    can access resources across all scopes and namespaces. Platform administrators should
+    have unrestricted access to manage any resource in the system, regardless of the
+    specific scope (organization, course, library, etc.).
+    """
+
+    POLICY = [
+        ["p", "role:platform_admin", "act:manage", "*", "allow"],
+        ["g", "user:user-1", "role:platform_admin", "*"],
+    ] + COMMON_ACTION_GROUPING
+
+    GENERAL_CASES = [
+        {
+            "subject": "user:user-1",
+            "action": "act:manage",
+            "scope": "*",
+            "expected_result": True,
+        },
+        {
+            "subject": "user:user-1",
+            "action": "act:manage",
+            "scope": "org:any-org",
+            "expected_result": True,
+        },
+        {
+            "subject": "user:user-1",
+            "action": "act:manage",
+            "scope": "course-v1:any-org+any-course+any-course-run",
```

Comment:
Should these scopes also need to be namespaced as well (like the org)? Same question for the other tests on course/lib scopes.